aws/resource_aws_ssm_parameter.go

```diff
@@ -147,6 +151,10 @@ func resourceAwsSsmParameterRead(d *schema.ResourceData, meta interface{}) error
 	}
 	d.Set("allowed_pattern", detail.AllowedPattern)
 
+	if detail.DataType != nil {
+		d.Set("data_type", detail.DataType)
+	}
```

Comment:
We generally prefer to always call `d.Set()` on attributes as it handles `nil` values automatically. 👍 

If Terraform starts reporting a difference for `String` type parameters (but not others), we can add `Computed: true` to the `data_type` schema.

```suggestion
	d.Set("data_type", detail.DataType)
```